File: mobile/lib/dtos/generate_cards_request.dart
class GenerateCardsRequest {
  String topic;
  int cardCount;

  GenerateCardsRequest({
    required this.topic,
    required this.cardCount,
  });

  factory GenerateCardsRequest.fromJson(Map<String, dynamic> json) =>
      GenerateCardsRequest(
        topic: json["topic"],
        cardCount: json["cardCount"],
      );

  Map<String, dynamic> toJson() => {
        "topic": topic,
        "cardCount": cardCount,
      };
}

File: backend/api/DTO/GenerateCardsRequest.cs
﻿using System.ComponentModel.DataAnnotations;

namespace api.DTO;

public class GenerateCardsRequest
{
    [Required]
    [MaxLength(100)]
    public string Topic { get; set; }
    [Required]
    [Range(1, 30)]
    public int CardCount { get; set; }
}
